variants/backend-base/config/initializers/check_env.rb

```diff
@@ -1,8 +1,42 @@
-# frozen-string-literal: true
+class VerifyPlaceholderSecretsNotUsedForReal
+  class << self
+    PLACEHOLDER_PREFIX_REGEX = /(PLACEHOLDER|FAILED_TO_GENERATE)/.freeze
 
-# RAILS_SECRET_KEY_BASE should be set to something other than its value in example.env
+    def run
+      return if local?
 
-if Rails.env.production? && Rails.root.join("example.env").read.include?(ENV.fetch("RAILS_SECRET_KEY_BASE"))
-  fail "RAILS_SECRET_KEY_BASE is unchanged from example.env. " \
-       "Generate a new one with `bundle exec rails secret`"
+      verify_secret_key_base
+      verify_activerecord_encryption_secrets
+    end
+
+    private
+
+    def verify_secret_key_base
+      return unless Rails.root.join("example.env").read.include?(ENV.fetch("RAILS_SECRET_KEY_BASE"))
+
+      fail "RAILS_SECRET_KEY_BASE is unchanged from example.env. Generate a new one with `bundle exec rails secret`"
+    end
+
+    ##
+    # Verify that placeholder values created by the Ackama rails template are
+    # not being used for real.
+    #
```

Comment:
🐘 this will get incorrectly rendered by IDEs otherwise

```suggestion
    # Verify that placeholder values created by the Ackama rails template are
    # not being used for real.
```